tuning/sft_trainer.py

```diff
@@ -286,23 +296,67 @@ def main(**kwargs):
         choices=["pt", "lora", None, "none"],
         default="pt",
     )
+    parser.add_argument(
+        "--exp_metadata",
+        type=str,
+        default=None,
+    )
     (
         model_args,
         data_args,
         training_args,
         lora_config,
         prompt_tuning_config,
-        peft_method,
+        aim_config,
+        additional,
         _,
     ) = parser.parse_args_into_dataclasses(return_remaining_strings=True)
-    if peft_method.peft_method == "lora":
-        tune_config = lora_config
-    elif peft_method.peft_method == "pt":
-        tune_config = prompt_tuning_config
+
+    peft_method = additional.peft_method
+    if peft_method =="lora":
+        tune_config=lora_config
+    elif peft_method =="pt":
+        tune_config=prompt_tuning_config
     else:
-        tune_config = None
-    train(model_args, data_args, training_args, tune_config)
+        tune_config=None
+
+    tracker_name = training_args.tracker
+    if tracker_name == "aim":
+        tracker_config=aim_config
+    else:
+        tracker_config=None
+
+    # Initialize callbacks
+    file_logger_callback = FileLoggingCallback(logger)
+    peft_saving_callback = PeftSavingCallback()
+    callbacks = [peft_saving_callback, file_logger_callback]
+
+    # Initialize the tracker
+    tracker = get_tracker(tracker_name, tracker_config)
```

Comment:
why not just have a `tracker_install_callbacks` function somewhere to reduce 3 lines into 1

Reply:
Agreed. Here too it's a design choice we have to make. We have tried to keep it consistent with how config is managed in fms-hf-tuning here too. This has tried to follow the pattern used in the peft config block above. But if we choose to have a different approach we can follow that.

Reply:
Its simple choice to either perform things explicitly or run the code in some other funciton reverting to no callback under the hood. Does not affect the functionality. If you strongly feeling a meta function can help we can make that change.